go-controller/pkg/ovn/policy.go

```diff
@@ -61,26 +142,46 @@ type networkPolicy struct {
 	policy          *knet.NetworkPolicy
 	ingressPolicies []*gressPolicy
 	egressPolicies  []*gressPolicy
+	isIngress       bool
+	isEgress        bool
 	podHandlerList  []*factory.Handler
 	svcHandlerList  []*factory.Handler
 	nsHandlerList   []*factory.Handler
 
-	// localPods is a list of pods affected by this policy
-	// this is a sync map so we can handle multiple pods at once
-	// map of string -> *lpInfo
+	// localPods is a map of pods affected by this policy.
+	// It is used to update defaultDeny port group port counters, when deleting network policy.
+	// Port should only be added here if it was successfully added to default deny port group,
+	// and local port group in db.
+	// This can be used by multiple local events for different objects in parallel,
+	// and retry_obj guarantees that the same key won't be handled by multiple threads.
+	// map of portName(string): portUUID(string)
 	localPods sync.Map
 
 	portGroupName string
 	deleted       bool //deleted policy
 }
 
 func NewNetworkPolicy(policy *knet.NetworkPolicy) *networkPolicy {
+	// Any pod that matches a network policy should get a default
```

Comment:
Actually this condition is wrong and @jcaamano  is fixing it https://github.com/ovn-org/ovn-kubernetes/pull/3162
Maybe it's better to include that behaviour here, since I already know the previous condition is wrong

Reply:
Or let's merge that PR first, and I will rebase